libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -1168,10 +1168,6 @@ void EthStratumClient::processResponse(Json::Value& responseObject)
                 diffToTarget((uint32_t*)m_nextWorkBoundary.data(), nextWorkDifficulty);
                 cnote << "Difficulty set to " EthWhite << nextWorkDifficulty
                       << EthReset " (nicehash)";
-                if (m_onWorkReceived && (m_current.epoch != -1))
```

Comment:
Eth stratum spec says that difficulty updates apply to the next job, not the current job. Do not resubmit the current job.